Packs/Zimperium/Integrations/ZimperiumV2/ZimperiumV2.py

```diff
@@ -0,0 +1,1034 @@
+    )
+    demisto.debug(f"fetching incidents between {start_time=} and {end_time=}, with {limit=}")
+
+    search_params = {key: value for param in fetch_query for key, value in [param.split('=', 1)]}
+    demisto.debug(f'The query for fetch: {search_params}')
+
+    res = client.threat_search(after=start_time, search_params=search_params, size=limit, sort=FETCH_FIELD)
+    incidents_res = res.get('content', [])
+    demisto.debug(f'Got {len(incidents_res)} incidents from the API, before filtering')
+
+    incidents_filtered = filter_incidents_by_duplicates_and_limit(
+        incidents_res=incidents_res,
+        last_run=last_run,
+        fetch_limit=max_fetch,
+        id_field='id'
+    )
+    demisto.debug(f'After filtering, there are {len(incidents_filtered)} incidents')
+
+    incidents: list[dict] = []
+    for incident in incidents_filtered:
+        occurred = timestamp_to_datestring(incident.get(FETCH_FIELD))
+        demisto.debug(f'Looking on: {incident.get("id")}, {occurred=}')
+        incident[FETCH_FIELD] = occurred
+        incidents.append({
+            'name': f"Threat {incident.get('id')} on Device ID {incident.get('deviceId')}",
+            'occurred': occurred,
+            'dbotMirrorId': incident.get('id'),
+            'severity': incident.get('severity'),
+            'rawJSON': json.dumps(incident)
+        })
+
+    last_run = update_last_run_object(
+        last_run=last_run,
+        incidents=incidents_filtered,
+        fetch_limit=max_fetch,
+        start_fetch_time=start_time,
+        end_fetch_time=end_time,
+        look_back=look_back,
+        created_time_field=FETCH_FIELD,
+        id_field='id',
+        date_format=DATE_FORMAT,
+        increase_last_run_time=False
+    )
+    demisto.debug(f"Last run after the fetch run: {last_run}")
+    return incidents, last_run
+
+
+def main():
+    params = demisto.params()
+    client_id = params.get('credentials', {}).get('identifier')
+    client_secret = params.get('credentials', {}).get('password')
+    base_url = urljoin(params.get('url'), '/api')
+    verify = not params.get('insecure', False)
+
+    # fetch params
+    max_fetch = arg_to_number(params.get('max_fetch', 50)) or 50
+    fetch_query = params.get('fetch_query') or ''
+    first_fetch = params.get('first_fetch', '7 days').strip()
+    look_back = arg_to_number(params.get('look_back')) or 1
+
+    first_fetch_time = arg_to_datetime(first_fetch)
+    first_fetch_time_str = first_fetch_time.strftime(DATE_FORMAT) if first_fetch_time else None
+
+    command = demisto.command()
+    args = demisto.args()
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(base_url=base_url, client_id=client_id, client_secret=client_secret, verify=verify)
+        if command == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            return_results(test_module(client, first_fetch_time_str))
```

Comment:
please test the fetch flow (including parameters) in the test module, if isFetch is true